### PR TITLE
feat: User facade get() function loads user details under the hood. (GH-2796)

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-2796

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/user/login/login.component.spec.ts [hunk 1]
```diff
@@ -111,21 +111,26 @@ describe('LoginComponent', () => {
 
   it('should have user details when token exists', () => {
     let user;
+    component.ngOnInit();
```
Comment (WeizhengSap) [inline projects/storefrontlib/src/cms-components/user/login/login.component.spec.ts]: maybe this `ngOnInit` can go to `beforeEach`.

File: projects/core/src/user/facade/user.service.spec.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ describe('UserService', () => {
     }
   ));
 
-  it('should be able to get user details', () => {
+  it('get ) should be able to get user details when they are present in the store', () => {
```
Comment (WeizhengSap) [inline projects/core/src/user/facade/user.service.spec.ts]: `get )` :)

File: projects/storefrontlib/src/cms-components/user/login/login.component.spec.ts (hunk 1 above)
Comment (plabadie) [reply projects/storefrontlib/src/cms-components/user/login/login.component.spec.ts]: Fixed.  However, `component.ngOnInit();` still needs to be called again after the spy statements.

File: projects/core/src/user/facade/user.service.spec.ts (hunk 2 above)
Comment (plabadie) [reply projects/core/src/user/facade/user.service.spec.ts]: Fixed.

**Pull request merged**